workflow-runtime/src/main/java/com/squareup/workflow/internal/WorkflowNode.kt

```diff
@@ -282,6 +308,16 @@ internal class WorkflowNode<PropsT, StateT, OutputT : Any, RenderingT>(
     return WorkerChildNode(worker, key, workerChannel, handler = handler)
   }
 
+  private fun createSideEffectNode(
+    key: String,
+    sideEffect: suspend () -> Unit
+  ): SideEffectNode {
+    // workerContext is guaranteed to not contain a Job.
+    val scope = this + CoroutineName("sideEffect[$key] for $id") + workerContext
```

Comment:
Coroutine name should override any name in worker context - switch order and add tests for name and also base/worker context order, including test for default worker dispatcher (switch to unconfined?) and overridden dispatcher.